src/types_.py

```diff
@@ -9,6 +9,25 @@
 from pathlib import Path
 from urllib.parse import urlparse
 
+
+class Metadata(typing.NamedTuple):
+    """Information within metadata file. Refer to: https://juju.is/docs/sdk/metadata-yaml.
+
+    Attrs:
+        name: Name of the charm.
+
+        docs: A link to a documentation cover page on Discourse.
+    """
+
+    # Required fields
```

Comment:
Is this implied by the type? And should this be in the docstring rather than as a comment?